ixmp/model/__init__.py

```diff
@@ -1,11 +1,36 @@
-#: Mapping from names to available models. To register additional models,
-#: add elements to this variable.
-MODELS: dict[str, type] = {}
+from typing import TYPE_CHECKING, Optional
 
+from .dantzig import DantzigModel
+from .gams import GAMSModel
 
-def get_model(name, **model_options):
-    """Return a model for *name* (or the default) with *model_options*."""
+if TYPE_CHECKING:
+    import ixmp.model.base
+
+#: Mapping from names to available models. To register additional models, add elements
+#: to this variable.
+MODELS: dict[str, type["ixmp.model.base.Model"]] = {
+    "default": GAMSModel,
+    "gams": GAMSModel,
+    "dantzig": DantzigModel,
+}
+
+
+def get_model(name: Optional[str], **model_options) -> "ixmp.model.base.Model":
+    """Return a model instance for `name` (or the default) with `model_options`.
+
+    Note that unlike :func:`.backend.get_class`, this function creates a new instance.
+
+    Parameters
+    ----------
+    name :
```

Comment:
```suggestion
    name : str, optional
```
Why not add the type hint here, too?

Reply:
In short, I think it's now redundant:
- Our Sphinx configuration result in the built docs displaying argument types that get derived from the type hints.
- A person reading the code itself can see the hints directly.